### PR TITLE
20930-Methods-using-printString-asString--or-storeString-in-printOnstoreOn

### PR DESCRIPTION
**Pull request by Ducasse**
redo and fix some of the change of badetitou

### CODE REVIEW

**Comment by svenvc** [comment]
I haven't  checked them all in details, but yes this make sense

File: src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st [hunk 1]
```diff
@@ -58,18 +58,40 @@ FreeTypeEmbeddedFileInfo >> locationType [
...
-		nextPutAll: ' - ', postscriptName asString;		
-		nextPutAll: ' ',(bold ifTrue:['B'] ifFalse:['']);
-		nextPutAll: ' ',(italic ifTrue:['I'] ifFalse:['']);
-		nextPutAll: ' ',(fixedWidth ifTrue:['Monospaced'] ifFalse:['']);
-		nextPutAll: ' ',(stretchValue asString);
-		nextPutAll: ' ',(weightValue asString);	
-		cr
+
+	aStream
+			nextPut: '{';
```
Comment (jecisc) [inline src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st]:
```suggestion
 			nextPut: ${;
```

File: src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st [hunk 2]
```diff
@@ -58,18 +58,40 @@ FreeTypeEmbeddedFileInfo >> locationType [
...
+			print: self locationType;
+			nextPut: '}';
```
Comment (jecisc) [inline src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st]:
```suggestion
 			nextPut: $};
```

File: src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st [hunk 3]
```diff
@@ -58,18 +58,40 @@ FreeTypeEmbeddedFileInfo >> locationType [
...
+			nextPut: '(';
```
Comment (jecisc) [inline src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st]:
```suggestion
 			nextPut: $(;
```

File: src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st [hunk 4]
```diff
@@ -58,18 +58,40 @@ FreeTypeEmbeddedFileInfo >> locationType [
...
+			print: fileContents size;
+			nextPutAll: ' bytes )';
+			nextPut: '[';
```
Comment (jecisc) [inline src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st]:
```suggestion
 			nextPut: $[;
```

File: src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st [hunk 5]
```diff
@@ -58,18 +58,40 @@ FreeTypeEmbeddedFileInfo >> locationType [
...
+			print: index;
+			nextPutAll: '] ';
+			print: familyName;
+			nextPutAll: ' - ';
+			print: styleName;
+			nextPutAll: ' - ';
+			print: postscriptName;
+			nextPutAll: ' ';
```
Comment (jecisc) [inline src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st]:
```suggestion
 			space;
```

File: src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st [hunk 6]
```diff
@@ -58,18 +58,40 @@ FreeTypeEmbeddedFileInfo >> locationType [
...
+			nextPutAll:
+				(bold
+					ifTrue: [ 'B' ]
+					ifFalse: [ '' ]);
+			nextPutAll: ' ';
```
Comment (jecisc) [inline src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st:82]:
```suggestion
 			space;
```

File: src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st [hunk 7]
```diff
@@ -58,18 +58,40 @@ FreeTypeEmbeddedFileInfo >> locationType [
...
+			nextPutAll:
+				(italic
+					ifTrue: [ 'I' ]
+					ifFalse: [ '' ]);
+			nextPutAll: ' ';
```
Comment (jecisc) [inline src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st:87]:
```suggestion
 			space;
```

File: src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st [hunk 8]
```diff
@@ -58,18 +58,40 @@ FreeTypeEmbeddedFileInfo >> locationType [
...
+			nextPutAll:
+				(fixedWidth
+					ifTrue: [ 'Monospaced' ]
+					ifFalse: [ '' ]);
+			nextPutAll: ' ';
```
Comment (jecisc) [inline src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st:92]:
```suggestion
 			space;
```

File: src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st [hunk 9]
```diff
@@ -58,18 +58,40 @@ FreeTypeEmbeddedFileInfo >> locationType [
...
+			print: stretchValue;
+			nextPutAll: ' ';
```
Comment (jecisc) [inline src/EmbeddedFreeType/FreeTypeEmbeddedFileInfo.class.st:94]:
```suggestion
 			space;
```

File: src/Monticello/MCWorkingCopy.class.st [hunk 10]
```diff
@@ -375,7 +375,10 @@ MCWorkingCopy >> possiblyNewerVersionsIn: aRepository [
 MCWorkingCopy >> printOn: aStream [
 
 	super printOn: aStream.
-	package name ifNotNil: [ aStream nextPutAll: '(' , package name asString, ')'].
+	package name ifNotNil: [ aStream 
+										nextPutAll: '(' ;
```
Comment (jecisc) [inline src/Monticello/MCWorkingCopy.class.st:379]:
```suggestion
 										nextPut: $( ; 
```

File: src/Monticello/MCWorkingCopy.class.st [hunk 11]
```diff
@@ -375,7 +375,10 @@ MCWorkingCopy >> possiblyNewerVersionsIn: aRepository [
 MCWorkingCopy >> printOn: aStream [
 
 	super printOn: aStream.
-	package name ifNotNil: [ aStream nextPutAll: '(' , package name asString, ')'].
+	package name ifNotNil: [ aStream 
+										nextPutAll: '(' ; 
+										print: package name ;
+										nextPutAll: ')']
```
Comment (jecisc) [inline src/Monticello/MCWorkingCopy.class.st]:
```suggestion
 										nextPut: $)]
```

File: src/Morphic-Base/HaloSpec.class.st [hunk 12]
```diff
@@ -45,10 +45,14 @@ HaloSpec >> iconSymbol [
...
 { #category : #printing }
 HaloSpec >> printOn: aStream [
-	"Add a textual printout representing the receiver to a stream"
 
 	super printOn: aStream.
-	aStream nextPutAll: ' (', addHandleSelector asString, ' ', iconSymbol asString, ')'
+	aStream 
+		nextPutAll: ' (';
+		print: addHandleSelector;
+		nextPutAll: ' ';
```
Comment (jecisc) [inline src/Morphic-Base/HaloSpec.class.st]:
```suggestion
 		space;
```

File: src/Morphic-Base/HaloSpec.class.st [hunk 13]
```diff
@@ -45,10 +45,14 @@ HaloSpec >> iconSymbol [
...
+		print: iconSymbol;
+		nextPutAll: ')'
```
Comment (jecisc) [inline src/Morphic-Base/HaloSpec.class.st]:
```suggestion
 		nextPut: $)
```

File: src/NECompletion/SHRange.class.st [hunk 14]
```diff
@@ -186,8 +186,14 @@ SHRange >> length [
 SHRange >> printOn: aStream [
 
 	super printOn: aStream. 
-	aStream nextPutAll: ' ', type. 
-	aStream nextPutAll: ' [', start asString, ' - ', end asString, ']'.
+	aStream 
+		nextPutAll: ' ';
```
Comment (jecisc) [inline src/NECompletion/SHRange.class.st]:
```suggestion
 		space;
```

File: src/NECompletion/SHRange.class.st [hunk 15]
```diff
@@ -186,8 +186,14 @@ SHRange >> length [
...
+		nextPutAll: type;
+		nextPutAll: ' [';
+		print: start ;
+		nextPutAll: ' - ';
+		print: end; 
+		nextPutAll: ']'
```
Comment (jecisc) [inline src/NECompletion/SHRange.class.st]:
```suggestion
 		nextPut: $]
```

**Review by jecisc — CHANGES_REQUESTED** [review]

**Comment by MarcusDenker** [comment]
it seems now we have lots of failing test in the released image

**Comment by MarcusDenker** [comment]
unix-32 / Tests-unix-32 / Unix32.FileSystem.Tests.Core.Base.PathTest.testRelativePrintString | 5 ms | 3
unix-32 / Tests-unix-32 / Unix32.Kernel.Tests.Extended.Classes.ClassTest.testAddClassSlot | 97 ms | 3
unix-32 / Tests-unix-32 / Unix32.NECompletion.Tests.NECEntryTest.testPrintOn | 9 ms | 3
unix-32 / Tests-unix-32 / Unix32.Refactoring.Tests.Core.RBRefactoringTest.testConditions | 14 ms | 3
unix-32 / Tests-unix-32 / Unix32.Slot.Tests.SlotClassVariableTest.testSlotIsPersistedAfterRebuildOfClass | 0.1 sec | 3
unix-32 / Tests-unix-32 / Unix32.Slot.Tests.SlotClassVariableTest.testAccessorInstanceVariableSlotGeneratesAccessors

**Comment by MarcusDenker** [comment]
maybe not all due to this but the printOn: looks related

**Pull request merged**